src/query/expression/src/type_check.rs

```diff
@@ -228,6 +231,34 @@ fn wrap_nullable_for_try_cast(span: Span, ty: &DataType) -> Result<DataType> {
     }
 }
 
+pub fn check_number<Index: ColumnIndex, T: Number>(
+    span: Span,
+    func_ctx: FunctionContext,
+    expr: &Expr<Index>,
+    fn_registry: &FunctionRegistry,
+) -> Result<T> {
+    let (expr, _) = ConstantFolder::fold(expr, func_ctx, fn_registry);
```

Comment:
just use evaluator and 'to_xxx'

Reply:
Or a CastExpr to `NumberType::<T>::data_type()`

Reply:
This needs to new a `Evaluator` and construct a `DataBlock`. I think using `ConstantFolder` is more convenient.